### PR TITLE
xds: support reporting loads for more than one cluster services

### PR DESCRIPTION
**Pull request by voidzcy**
The LRS implementation was kind of half-baked. We have APIs for adding `LoadStatsStore`s as source of load data (per cluster service) to be reported, but we expect management server to ask load reporting for only one cluster service. This doesn't make sense. 

The assumption for management server to ask load reporting for only one cluster service seems overly strong. It is not consistent with our LRS client's design that it is capable of _reporting loads for multiple services of a cluster_. The reality is that we still will only have at most one service per cluster for now, but management server may ask loads for multiple services even if rest are completely unknown or not used to the client. 

The most correct way is to support reporting loads for multiple services. The actual loads to be reported will be the intersection of services that we have loads for and services that management server asks for.

This PR also cleans up tests for LRS client's implementation a bit. It becomes less verbose and focus on testing LRS client's major behaviors.

### CODE REVIEW

File: xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java [hunk 1]
```diff
@@ -474,81 +346,6 @@ public void lrsStreamClosedAndRetried() {
     verify(backoffPolicy2, times(1)).nextBackoffNanos();
   }
 
-  @Test
-  public void lrsStreamRetryAndRereport() {
```
Comment (dapengzhang0) [inline xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: Why is this removed? Is there behavior change?

File: xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java [hunk 2]
```diff
@@ -186,201 +189,69 @@ public void tearDown() {
...
-
-  // Currently we expect each gRPC client talks to a single service per cluster, so we test LRS
-  // client reporting load for a single cluster service only.
-  // TODO(chengyuanzhang): Existing test suites for LRS client implementation have poor behavior
-  //  coverage and are not robust. Should improve once its usage is finalized without too much
-  //  assumption.
-
-  @Test
-  public void loadReportActualIntervalAsSpecified() {
+  public void normalOperations() {
```
Comment (dapengzhang0) [inline xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: The name does not provide any information what to test and what to verify. Either give it a more descriptive name or add a description of what are being tested in javadoc.

File: xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java (hunk 1 above)
Comment (voidzcy) [reply xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: This test case was kind of redundant as all it wanted to show is LRS is back to normal after stream closed and retry. Not too much value to test this, it's verbose.

File: xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java (hunk 2 above)
Comment (voidzcy) [reply xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: Hmm... This indeed tests LRS client's normal operation, basically the periodic load reporting. Previously, it was splited in multiple test cases such as initial request/response, load reporting interval update, etc. It is verbose to have them be separate as usually one behavior is just a step further from another, and those test cases have significant code repeat. So I think it would be better to put them together, which is just a normal flow of LRS client. Nothing specific, normal workflow is what it is testing.

File: xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java (hunk 1 above)
Comment (dapengzhang0) [reply xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: I think it doesn't harm if an exiting test has some redundancy and verbose. Is the re-report part covered elsewhere?

Comment (voidzcy) [reply xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: We can extend the retry test case a little bit to show the re-report, so that we do not need to repeat the logic for stream closed and retry.

Comment (voidzcy) [reply xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: Added.

File: xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java (hunk 2 above)
Comment (dapengzhang0) [reply xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: The maybe `normalLrsFlow()` is a better name.

Comment (voidzcy) [reply xds/src/test/java/io/grpc/xds/LoadReportClientImplTest.java]: `typicalWorkflow`. I saw there is a similar one in [`HealthCheckingLoadBalancerFactoryTest`](https://github.com/grpc/grpc-java/blob/659987185f9369f6d53edd2d9cfab2461409f910/services/src/test/java/io/grpc/services/HealthCheckingLoadBalancerFactoryTest.java#L251).
